src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisObserver.java

```diff
@@ -7,63 +7,32 @@
 @Service
-public class SensitivityAnalysisObserver {
-
-    private static final String OBSERVATION_PREFIX = "app.computation.";
-    private static final String PROVIDER_TAG_NAME = "provider";
-    private static final String TYPE_TAG_NAME = "type";
-    private static final String STATUS_TAG_NAME = "status";
+public class SensitivityAnalysisObserver extends AbstractComputationObserver<Void, SensitivityAnalysisInputData> {
     private static final String COMPUTATION_TYPE = "sensi";
-    private static final String COMPUTATION_COUNTER_NAME = OBSERVATION_PREFIX + "count";
-
-    private final String defaultProvider;
-    private final ObservationRegistry observationRegistry;
-    private final MeterRegistry meterRegistry;
 
-    public SensitivityAnalysisObserver(@Value("${sensitivity-analysis.default-provider}") String defaultProvider,
-                                       @NonNull ObservationRegistry observationRegistry,
+    public SensitivityAnalysisObserver(@NonNull ObservationRegistry observationRegistry,
                                        @NonNull MeterRegistry meterRegistry) {
-        this.defaultProvider = defaultProvider;
-        this.observationRegistry = observationRegistry;
-        this.meterRegistry = meterRegistry;
-    }
-
-    public <E extends Throwable> void observe(String name, SensitivityAnalysisRunContext runContext, Observation.CheckedRunnable<E> runnable) throws E {
-        createObservation(name, runContext).observeChecked(runnable);
-    }
-
-    public <T, E extends Throwable> T observeRun(String name, SensitivityAnalysisRunContext runContext, Observation.CheckedCallable<T, E> callable) throws E {
-        T result = createObservation(name, runContext).observeChecked(callable);
-        incrementCount(runContext, result);
-        return result;
+        super(observationRegistry, meterRegistry);
     }
 
-    private Observation createObservation(String name, SensitivityAnalysisRunContext runContext) {
-        String provider = runContext.getProvider() != null ? runContext.getProvider() : defaultProvider;
-        return Observation.createNotStarted(OBSERVATION_PREFIX + name, observationRegistry)
-            .lowCardinalityKeyValue(PROVIDER_TAG_NAME, provider)
-            .lowCardinalityKeyValue(TYPE_TAG_NAME, COMPUTATION_TYPE);
+    @Override
+    protected String getComputationType() {
+        return COMPUTATION_TYPE;
     }
 
-    private <T> void incrementCount(SensitivityAnalysisRunContext runContext, T result) {
-        String provider = runContext.getProvider() != null ? runContext.getProvider() : defaultProvider;
-        Counter.builder(COMPUTATION_COUNTER_NAME)
-            .tag(PROVIDER_TAG_NAME, provider)
-            .tag(TYPE_TAG_NAME, COMPUTATION_TYPE)
-            .tag(STATUS_TAG_NAME, result == null ? "NOK" : "OK")
-            .register(meterRegistry)
-            .increment();
+    @Override
+    protected String getResultStatus(Void res) {
+        return res == null ? "NOK" : "OK";
```

Comment:
Will always return "NOK" => is there a way to improve this ?

Reply:
Discussed with Joris : it is better to always return "OK". If there is an error and a throw it will be caught and displayed in grafan as an error through an other function.

We can consider that if the computation reach "getResultStatus" withou a throw, it means that the computation worked.

It may be improved later with an other PR.